test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -2531,79 +2535,144 @@ class test_ucp_sockaddr_protocols_err_sender
+     * if sender EP was killed right after sent RTS then receiver may get
+     * incorrect/corrupted data */
+    void do_tag_rndv_killed_sender_test(bool reg_mem)
+    {
+        static const size_t size = 64 * UCS_KBYTE;
+
+        /* Warmup */
+        test_tag_send_recv(size, false);
+        request_wait(sender().flush_worker_nb());
+        request_wait(receiver().flush_worker_nb());
+
+        /* Check for INVALIDATE only after warmup to ensure that the final
+         * configuration was selected on UCP endpoint */
+        bool has_invalidate             = false;
+        ucp_ep_config_key_t *config_key = &ucp_ep_config(sender().ep())->key;
+        for (unsigned i = 0;
+             (i < UCP_MAX_LANES) &&
+             (config_key->rma_bw_lanes[i] != UCP_NULL_LANE); i++) {
+            ucp_lane_index_t lane              = config_key->rma_bw_lanes[i];
+            const uct_md_attr_t *md_attr       =
+                    ucp_ep_md_attr(sender().ep(), lane);
+            const uct_iface_attr_t *iface_attr =
+                    ucp_ep_get_iface_attr(sender().ep(), lane);
+
+            if ((md_attr->cap.flags & UCT_MD_FLAG_INVALIDATE) &&
+                (iface_attr->cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE)) {
+                has_invalidate = true;
+            }
+        }
 
-/* This test is quite tricky: it checks for incorrect behavior on RNDV send
- * on DC transport: in case if sender EP was killed right after sent RTS
- * then receiver may get incorrect/corrupted data */
-UCS_TEST_P(test_ucp_sockaddr_protocols_err_sender, tag_rndv_killed_sender,
-           "RNDV_THRESH=10k", "RNDV_SCHEME=get_zcopy")
-{
-    static size_t size = 64 * UCS_KBYTE;
-    static const std::string dc_tls[] = { "dc", "dc_x" };
-    bool has_dc = has_any_transport(
-        std::vector<std::string>(dc_tls,
-                                 dc_tls + ucs_static_array_size(dc_tls)));
+        if (!has_invalidate) {
+            UCS_TEST_SKIP_R("Has no RNDV transport with INVALIDATE support");
+        }
 
-    if (!has_dc)
-    {
-        UCS_TEST_SKIP_R("Unsupported");
+        std::string send_buf(size, 'x');
+        std::string recv_buf(size, 'y');
+        std::string recv_copy(size, 'y');
+ 
+        ucp_mem_h ucp_memh = NULL;
+        ucp_tag_message_h message;
+        ucp_tag_recv_info_t info;
+        ucs_status_t status;
+
+        if (reg_mem) {
+            /* Do additional memory registration of the same send buffer to
+             * test a case that the memory validation would not be completed
+             * immediately when UCP endpoint is destroyed and all UCP requests
+             * are purged */
+            ucp_mem_map_params_t params = {0};
+
+            params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
+                                UCP_MEM_MAP_PARAM_FIELD_LENGTH;
+            params.address    = &send_buf[0];
+            params.length     = size;
+
+            ucs_status_t status = ucp_mem_map(sender().ucph(), &params,
```

Comment:
we plan to make ucp_mem_map not delay request completion. can we make the test not rely on ucp_mem_map?
@gleon99

Reply:
done